### PR TITLE
Profiling of LuaJIT GC

### PR DESCRIPTION
**Pull request by themuflon**

### CODE REVIEW

**Comment by bengarney** [comment]
```
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:715:3: warning: implicit declaration of function 'LUA_GC_BEGIN' is invalid in C99 [-Wimplicit-function-declaration]
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   LUA_GC_BEGIN(step)
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:715:16: error: use of undeclared identifier 'step'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   LUA_GC_BEGIN(step)
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:718:20: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   int32_t ostate = g->vmstate;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                    ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:719:14: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   setvmstate(g, GC);
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]              ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] ./lj_obj.h:441:29: note: expanded from macro 'setvmstate'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] #define setvmstate(g, st)       ((g)->vmstate = ~LJ_VMST_##st)
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                                   ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:720:28: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   lim = (GCSTEPSIZE/100) * g->gc.stepmul;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                            ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:723:3: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   g->gc.debt += g->gc.total - g->gc.threshold;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:723:17: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   g->gc.debt += g->gc.total - g->gc.threshold;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                 ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:723:31: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   g->gc.debt += g->gc.total - g->gc.threshold;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                               ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:726:9: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     if (g->gc.state == GCSpause) {
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]         ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:727:7: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       g->gc.threshold = (g->gc.estimate/100) * g->gc.pause;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:727:26: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       g->gc.threshold = (g->gc.estimate/100) * g->gc.pause;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                          ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:727:48: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       g->gc.threshold = (g->gc.estimate/100) * g->gc.pause;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                                                ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:728:7: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       g->vmstate = ostate;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:729:7: warning: implicit declaration of function 'LUA_GC_END' is invalid in C99 [-Wimplicit-function-declaration]
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       LUA_GC_END(step)
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:729:18: error: use of undeclared identifier 'step'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       LUA_GC_END(step)
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                  ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:733:7: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]   if (g->gc.debt < GCSTEPSIZE) {
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:734:5: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     g->gc.threshold = g->gc.total + GCSTEPSIZE;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:734:23: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     g->gc.threshold = g->gc.total + GCSTEPSIZE;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:736:5: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     g->gc.debt -= GCSTEPSIZE;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:737:5: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     g->gc.threshold = g->gc.total;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] lj_gc.c:737:23: error: use of undeclared identifier 'g'
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]     g->gc.threshold = g->gc.total;
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a]                       ^
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] fatal error: too many errors emitted, stopping now [-ferror-limit=]
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] 2 warnings and 20 errors generated.
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] make[1]: *** [lj_gc.o] Error 1
[16:37:57][Execute build/luajit_x86/lib/libluajit-5.1.a] make[1]: *** Waiting for unfinished jobs....
[16:38:00][Execute build/luajit_x86/lib/libluajit-5.1.a] make: *** [buildOSX] Error 2
[16:38:00][Execute build/luajit_x86/lib/libluajit-5.1.a] 
[16:38:00]
[Execute build/luajit_x86/lib/libluajit-5.1.a] RuntimeError: Command failed with status (2): [make b
```

File: Rakefile [hunk 1]
```diff
@@ -36,6 +36,9 @@ end
 # If 1, then we link against LuaJIT. If 0, we use classic Lua VM.
 $doBuildJIT=1
 
+# If 1, then LUA GC profiling code is enabled
+$doEnableLuaGcProfile=0
```
Comment (bengarney) [inline Rakefile]: This should be tied to the debug/release status of the build, not hardcoded.

**Comment by bengarney** [comment]
Nitted!

File: Rakefile (hunk 1 above)
Comment (themuflon) [inline Rakefile]: Yes, but this can easily be done in Rakefile

Comment (bengarney) [inline Rakefile]: Yes -  I mean this flag's value should be based on the value here: https://github.com/LoomSDK/LoomSDK/blob/JG-LuaJIT-Profiling/Rakefile#L11

**Pull request merged**